testing/agent-exporter/agent-exporter.gradle

```diff
@@ -14,22 +15,10 @@ dependencies {
   annotationProcessor deps.autoservice
   compileOnly deps.autoservice
 
-  implementation project(':javaagent-tooling')
```

Comment:
I had to move this to allow its SPIs to be merged with instrumentations. It made maintaining the agent-for-testing test difficult - I just deleted it since while it was a nice sanity check while getting that testing infrastructure in place, now it doesn't really provide much value.